### DISCUSSION TITLE
How best to store Legacy Token (Token V1) Identifiers?

### DISCUSSION THREAD

**Discussion by gregnazario** [aptos-labs/aptos-developer-discussions#92] [Questions] [answered]
From Discord:

> My question would be, what is the best way to keep an id of a token_v1 collection? With v2 it's easy as there is an Object<Collection>, but for V1, they are generally defined by the creator and name.
> So for now, what I do is just make a vector<vector<u8>> and use that as an Id, as i need the collection id for the key in a table

**Comment by gregnazario** [comment, marked as answer]
## Storing Legacy Token(Token V1) Collection Ids

Since Legacy tokens collections are referenced by creator, and collection name, you'll just need to store both.

You can create a struct like below:

```move
use aptos_framework::string::String;

struct LegacyCollectionId has store, drop {
  creator: address,
  name: String
}

struct Storage has key {
  legacy_collections: vector<LegacyCollectionId>
}
```


## Storing Legacy Token(Token V1) Token Ids

For tokens, you can store the `TokenDataId` or the `TokenId` based on your needs:

```move
vector<TokenId> ...
```

https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/aptos-token/sources/token.move#L168-L184

**Reply by unbound26z** [reply]
Thanks again for this answer!
